examples/basic-event-handlers/test/handlers.js

```diff
@@ -0,0 +1,110 @@
+const path = require('path')
+const { system, patching } = require('gluegun')
+const { createApolloFetch } = require('apollo-fetch')
+
+const GravatarRegistry = artifacts.require('./GravatarRegistry.sol')
+
+const srcDir = path.join(__dirname, '..')
+
+const fetchSubgraphs = createApolloFetch({ uri: 'http://localhost:8030/graphql' })
+const fetchSubgraph = createApolloFetch({
+  uri: 'http://localhost:8000/subgraphs/name/test/basic-event-handlers',
+})
+
+const waitForSubgraphToBeSynced = async () =>
```

Comment:
This function looks good. But I wonder, what if someone is trying to sync a subgraph that is going to take hours or days? and they want to test it.

I'm thinking two things. One - we allow them to point to an already fully synced subgraph.

Or if we want to test specific handlers on early blocks , we could designate a block, like 8,000,000, and then just test the first time each handler runs. 

I guess this is only relevant if we are allowing testing on mainnet. Since ganache will be very easy to sync

Reply:
Yeah. Again, this just an example subgraph with a basic test.

We may want to provide more convenient ways to wait for subgraphs to reach a certain block or until they sync completely, to query subgraphs etc. over time. This could be in a `@graphprotocol/graph-test` companion library or similar. For now I'm fine if all this is a bit raw.

If people wanted to test a subgraph syncing from a specific block, all they'd have to do as part of the test setup is to define a manifest with a start block.